### PR TITLE
Add README.md description

### PR DESCRIPTION
**Pull request by DieHard073055**

### CODE REVIEW

**Review by pamdayne — APPROVED** [review]

**Review by pamdayne — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1 +1,111 @@
-# supabase-hello-world
\ No newline at end of file
+
+# Supabase Hello World
+
+This project serves as a simple 'Hello World' example to illustrate how to connect to a Supabase database, perform an upsert operation to insert default data, and execute a select operation to retrieve all the data.
+
+## Code Explanation
+
+The `connection.js` file contains the main logic of the application. Here's a breakdown of its content:
```
Comment (pamdayne) [inline README.md:28]: Will `connection.js` be the only file for this whole repo?

File: README.md [hunk 2]
```diff
@@ -1 +1,111 @@
-# supabase-hello-world
\ No newline at end of file
+
+# Supabase Hello World
+
+This project serves as a simple 'Hello World' example to illustrate how to connect to a Supabase database, perform an upsert operation to insert default data, and execute a select operation to retrieve all the data.
+
+## Code Explanation
+
```
Comment (pamdayne) [inline README.md:20]:
```suggestion
### Importing modules
```

File: README.md [hunk 3]
```diff
@@ -1 +1,111 @@
...
+require('dotenv').config()
+const supabase = createClient(
+  process.env.SUPABASE_URL,
+  process.env.SUPABASE_ANON_KEY
+)
+
+const table_name = "countries"
+```
+
```
Comment (pamdayne) [inline README.md]:
```suggestion
### Inserting
```

File: README.md [hunk 4]
```diff
@@ -1 +1,111 @@
...
+      ])
+  if (error) {
+    console.error(error)
+  }
+  if (data) {
+    console.log(data)
+  }
+}
+```
+The `select_all_data` function retrieves all the records from the 'countries' table and logs them to the console. Errors, if any, are also logged to the console.
```
Comment (pamdayne) [inline README.md:70]:
```suggestion

### Selecting
The `select_all_data` function retrieves all the records from the 'countries' table and logs them to the console. Errors, if any, are also logged to the console.
```

File: README.md [hunk 5]
```diff
@@ -1 +1,111 @@
...
+
+  if (error) {
+    console.error(error)
+  }
+  if (data) {
+    console.log(data)
+  }
+}
+```
+Finally, we call the `select_all_data` function inside the main function to start the application.
```
Comment (pamdayne) [inline README.md:87]:
```suggestion
### Start using your application
Finally, we call the `select_all_data` function inside the main function to start the application.
```

**Review by pamdayne — COMMENTED** [review]
Some comment suggestions

File: README.md (hunk 1 above)
Comment (DieHard073055) [reply README.md:28]: No, we will break the code down to different as we see fit.
Supabase is capable of doing alot. 
As we document it we will group them together in a way that makes sense.

File: README.md [hunk 6]
```diff
@@ -1 +1,149 @@
...
+
+# Supabase Hello World
+
+This project serves as a simple 'Hello World' example to illustrate how to connect to a Supabase database, perform an upsert operation to insert default data, and execute a select operation to retrieve all the data.
+
+## Usage Instructions
+
+### Installation
+
+1. Install supabase using npm. This is available via [NPM](https://www.npmjs.com) as dev dependency.
```
Comment (pamdayne) [inline README.md]:
```suggestion
Install `supabase` using npm. 
This is available via [NPM](https://www.npmjs.com) as dev dependency.
```

File: README.md [hunk 7]
```diff
@@ -1 +1,150 @@
...
+
+### Installation
+
+Install `supabase` using npm. 
+This is available via [NPM](https://www.npmjs.com) as dev dependency.
+
+```bash
+npm i supabase --save-dev
+```
+### To run:
```
Comment (pamdayne) [inline README.md]:
```suggestion

Alternatively, if you're using macOS, supabase is available via Homebrew.
```bash
brew install supabase/tap/supabase```

### To run:

```

File: README.md [hunk 8]
```diff
@@ -1 +1,150 @@
...
+```bash
+npx supabase -h
+```
+Alternatively, if you're using macOS, supabase is available via Homebrew.
+
+```bash
+brew install supabase/tap/supabase
+```
```
Comment (pamdayne) [inline README.md]:
```suggestion

```

Comment (pamdayne) [reply README.md]: Move this into the `Installation` section.

File: README.md [hunk 9]
```diff
@@ -1 +1,151 @@
...
+
+```bash
+npm i supabase --save-dev
+```
+
+Alternatively, if you're using macOS, supabase is available via Homebrew.
+```bash
+brew install supabase/tap/supabase```
+
+### To run:
```
Comment (pamdayne) [inline README.md]:
```suggestion
Alternatively, if you're using macOS, supabase is available via Homebrew.
```bash
brew install supabase/tap/supabase
```

### To run:
```

File: README.md [hunk 10]
```diff
@@ -1 +1,149 @@
...
+This is available via [NPM](https://www.npmjs.com) as dev dependency.
+
+```bash
+npm i supabase --save-dev
+```
+
+Alternatively, if you're using macOS, supabase is available via Homebrew.
+```bash
+brew install supabase/tap/supabase
+
```
Comment (pamdayne) [inline README.md]:
```suggestion
## To run:
```

File: README.md [hunk 11]
```diff
@@ -1 +1,149 @@
...
+npm i supabase --save-dev
+```
+
+Alternatively, if you're using macOS, supabase is available via Homebrew.
+```bash
+brew install supabase/tap/supabase
+## To run:
+```bash
+npx supabase -h
+```
```
Comment (pamdayne) [inline README.md:23]:
```suggestion
	```bash
	brew install supabase/tap/supabase
	```

### To run:
	```bash
	npx supabase -h
	```
```

Comment (pamdayne) [inline README.md]:
```suggestion
	```bash
	brew install supabase/tap/supabase
	```

### To run:
	```bash
	npx supabase -h
	```
```

**Review by pamdayne — APPROVED** [review]

**Pull request merged**